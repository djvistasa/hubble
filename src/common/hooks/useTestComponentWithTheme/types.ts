type IUseTestComponentWithTheme = {};

export type { IUseTestComponentWithTheme };
