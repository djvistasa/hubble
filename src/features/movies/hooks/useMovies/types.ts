import { UseMutationResult } from "react-query";

type IUseMoviesReturnValues = {
  getNextPage: () => void;
  searchForMovie: UseMutationResult<
    never[] | undefined,
    unknown,
    string,
    unknown
  >;
  getMovies: () => void;
};

type IMoviesProps = {};

type IMoviesRouteParams = {
  page: string;
};

type ISearchParams = {
  total_pages: string;
};

export type {
  IMoviesProps,
  IMoviesRouteParams,
  ISearchParams,
  IUseMoviesReturnValues,
};
