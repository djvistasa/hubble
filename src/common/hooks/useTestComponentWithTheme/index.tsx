import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";

import { RenderResult } from "@testing-library/react";

import { ReactElement } from "react";
import { QueryClient, QueryClientProvider } from "react-query";

function useTestComponentWithTheme() {
  const renderWithTheme = (component: ReactElement<any, any>): RenderResult => {
    const queryClient = new QueryClient();
    return render(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </QueryClientProvider>
    );
  };

  return renderWithTheme;
}

export default useTestComponentWithTheme;
