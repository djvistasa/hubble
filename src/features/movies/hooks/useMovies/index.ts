import { IMovieResponse } from "@features/movies/types";
import { makeApiRequest } from "@utils";
import { useQuery } from "react-query";
import { useNavigate, useParams, useSearchParams } from "react-router-dom";
import { IMoviesRouteParams, IUseMovies } from "./types";

function useMovies(): IUseMovies {
  // create a getMovies mutation

  const { page } = useParams<IMoviesRouteParams>();
  const [searchParams, setSearchParams] = useSearchParams();

  const navigation = useNavigate();

  console.log(searchParams);

  const getNextPage = () => {
    if (page) {
      const activePage = parseInt(page);
      navigation(`/movies/${activePage + 1}`);
    }
  };

  const { data: movies } = useQuery(["movies"], async () => {
    const { error, result, ok } = await makeApiRequest(
      `/discover/movie?include_adult=false&include_video=false&language=en-US&page=${page}&sort_by=popularity.desc`
    );
    if (error) {
      // handle error
      return [];
    }

    if (result && ok) {
      const { results, total_pages } = result as IMovieResponse;

      setSearchParams({
        total_pages: total_pages.toString(),
      });

      return results;
    }
  });

  return { movies, getNextPage };
}

export default useMovies;
