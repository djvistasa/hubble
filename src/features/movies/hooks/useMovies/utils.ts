import { useMoviesStore } from "@features/movies/store";
import { IMovieResponse } from "@features/movies/types";
import { makeApiRequest } from "@utils";
import { SetURLSearchParams } from "react-router-dom";

const getPaginatedMoviesData = async (
  page: number,
  setSearchParams: SetURLSearchParams
) => {
  const { error, result, ok } = await makeApiRequest(
    `/discover/movie?include_adult=false&include_video=false&language=en-US&page=${page}&sort_by=popularity.desc`
  );

  if (error) {
    // handle error
    return [];
  }

  if (result && ok) {
    const { results, total_pages } = result as IMovieResponse;

    setSearchParams({
      total_pages: total_pages.toString(),
    });

    useMoviesStore.setState({ movies: results });
    useMoviesStore.setState({ filteredMovies: results });
  }
};

export { getPaginatedMoviesData };
