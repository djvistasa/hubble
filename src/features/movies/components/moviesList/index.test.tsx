import useTestComponentWithTheme from "@hooks/useTestComponentWithTheme";
import { describe, expect, it, vi } from "vitest";

import { moviesResp } from "@features/movies/hooks/useMovies/mocks";
import MoviesList from ".";

describe("MoviesList", () => {
  const mountWithTheme = useTestComponentWithTheme();
  const mockOnClick = vi.fn();

  const mockMovies = moviesResp.results;

  it("should render correctly", () => {
    const component = mountWithTheme(
      <MoviesList movies={mockMovies} onMovieClick={mockOnClick} />
    );
    expect(component).toMatchSnapshot();
  });

  it("should call onMovieClick when clicked", () => {
    const { getByTestId } = mountWithTheme(
      <MoviesList movies={mockMovies} onMovieClick={mockOnClick} />
    );
    const moviesList = getByTestId("movie-card-Sixty Minutes");
    moviesList.click();
    expect(mockOnClick).toHaveBeenCalledWith(mockMovies[1]);
  });
});
