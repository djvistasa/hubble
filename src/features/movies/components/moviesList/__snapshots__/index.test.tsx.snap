// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MoviesList > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__StyledMoviesList-sc-1n8iwya-0 hDFrCA"
      >
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 gZnIKa"
          data-testid="movie-card-Badland Hunters"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Badland Hunters
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 fHoZhD"
          data-testid="movie-card-Sixty Minutes"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Sixty Minutes
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 fOTeeA"
          data-testid="movie-card-Wonka"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Wonka
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 hVryrO"
          data-testid="movie-card-The Marvels"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            The Marvels
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 kcLDxU"
          data-testid="movie-card-One More Shot"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            One More Shot
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 dFboWy"
          data-testid="movie-card-The Family Plan"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            The Family Plan
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 idVGmy"
          data-testid="movie-card-Skal - Fight for Survival"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Skal - Fight for Survival
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 jYWfiQ"
          data-testid="movie-card-Lift"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Lift
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 bcbsKY"
          data-testid="movie-card-Madame Web"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Madame Web
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 eAeUta"
          data-testid="movie-card-Migration"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Migration
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 jkOUru"
          data-testid="movie-card-The Beekeeper"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            The Beekeeper
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 krvWOW"
          data-testid="movie-card-Aquaman and the Lost Kingdom"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Aquaman and the Lost Kingdom
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 eTanPM"
          data-testid="movie-card-Wish"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Wish
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 kWiAsi"
          data-testid="movie-card-The Hunger Games: The Ballad of Songbirds & Snakes"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            The Hunger Games: The Ballad of Songbirds & Snakes
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 cfPCLm"
          data-testid="movie-card-Silent Night"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Silent Night
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 knLBgZ"
          data-testid="movie-card-Rebel Moon - Part One: A Child of Fire"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Rebel Moon - Part One: A Child of Fire
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 bVFOzJ"
          data-testid="movie-card-Anyone But You"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Anyone But You
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 deWxus"
          data-testid="movie-card-Orion and the Dark"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Orion and the Dark
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 jRMdcE"
          data-testid="movie-card-Role Play"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            Role Play
          </p>
        </a>
        <a
          class="styles__StyledMovieCard-sc-vuoc3k-0 hGixLw"
          data-testid="movie-card-The Bricklayer"
        >
          <div
            class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
          />
          <p
            class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
          >
            The Bricklayer
          </p>
        </a>
      </div>
      <div
        class="styles__StyledPaginationWrapper-sc-1n8iwya-1 gTDeZP"
      >
        <div
          class="styles__StyledPagination-sc-kafszj-0 cHUXYz"
        >
          <button
            class="styles__StyledButton-sc-1mujxr2-0 iqLpCy"
            disabled=""
          >
            Previous
          </button>
          <button
            class="styles__StyledButton-sc-1mujxr2-0 iqLpCy"
            disabled=""
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__StyledMoviesList-sc-1n8iwya-0 hDFrCA"
    >
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 gZnIKa"
        data-testid="movie-card-Badland Hunters"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Badland Hunters
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 fHoZhD"
        data-testid="movie-card-Sixty Minutes"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Sixty Minutes
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 fOTeeA"
        data-testid="movie-card-Wonka"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Wonka
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 hVryrO"
        data-testid="movie-card-The Marvels"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          The Marvels
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 kcLDxU"
        data-testid="movie-card-One More Shot"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          One More Shot
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 dFboWy"
        data-testid="movie-card-The Family Plan"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          The Family Plan
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 idVGmy"
        data-testid="movie-card-Skal - Fight for Survival"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Skal - Fight for Survival
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 jYWfiQ"
        data-testid="movie-card-Lift"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Lift
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 bcbsKY"
        data-testid="movie-card-Madame Web"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Madame Web
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 eAeUta"
        data-testid="movie-card-Migration"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Migration
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 jkOUru"
        data-testid="movie-card-The Beekeeper"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          The Beekeeper
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 krvWOW"
        data-testid="movie-card-Aquaman and the Lost Kingdom"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Aquaman and the Lost Kingdom
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 eTanPM"
        data-testid="movie-card-Wish"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Wish
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 kWiAsi"
        data-testid="movie-card-The Hunger Games: The Ballad of Songbirds & Snakes"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          The Hunger Games: The Ballad of Songbirds & Snakes
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 cfPCLm"
        data-testid="movie-card-Silent Night"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Silent Night
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 knLBgZ"
        data-testid="movie-card-Rebel Moon - Part One: A Child of Fire"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Rebel Moon - Part One: A Child of Fire
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 bVFOzJ"
        data-testid="movie-card-Anyone But You"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Anyone But You
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 deWxus"
        data-testid="movie-card-Orion and the Dark"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Orion and the Dark
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 jRMdcE"
        data-testid="movie-card-Role Play"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          Role Play
        </p>
      </a>
      <a
        class="styles__StyledMovieCard-sc-vuoc3k-0 hGixLw"
        data-testid="movie-card-The Bricklayer"
      >
        <div
          class="styles__StyledMask-sc-vuoc3k-2 LYXKU mask"
        />
        <p
          class="styles__StyledMovieInfoWrapper-sc-vuoc3k-1 gKOtGX"
        >
          The Bricklayer
        </p>
      </a>
    </div>
    <div
      class="styles__StyledPaginationWrapper-sc-1n8iwya-1 gTDeZP"
    >
      <div
        class="styles__StyledPagination-sc-kafszj-0 cHUXYz"
      >
        <button
          class="styles__StyledButton-sc-1mujxr2-0 iqLpCy"
          disabled=""
        >
          Previous
        </button>
        <button
          class="styles__StyledButton-sc-1mujxr2-0 iqLpCy"
          disabled=""
        >
          Next
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
